designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/BaseEndpointDefinitions.scala

```diff
@@ -67,37 +66,20 @@ object BaseEndpointDefinitions {
                   )
                 )
             ),
-            oneOfVariantFromMatchType(
-              NotFound,
-              plainBody[ImpersonatedUserDataNotFoundError.type]
```

Comment:
I've changed 404 to 403 for `ImpersonatedUserDataNotFoundError`. 
1. `withSecurity` is added to different endpoint definitions which also can have 404 code declared - in such a case we override some fields and we also somehow breaking openApi spec (two times 404 code defined for one endpoint) 
2. Personally I think that `withSecurity` along with unsuccessful impersonation is not 404 but 403. Should `withSecurity ever return 404? :thinking:

Reply:
Now I think I could change this error to `ImpersonatedUserDataNotExists` - WDYT?

Reply:
the current implementation makes sense for me. the `ImpersonatedUserDataNotFoundError` means that the user was able to authenticate, but the impersonated user was not found. There was no authorization issue. 

But if it simplifies the tapir things, we can return 403 instead.

According to the name of the error: I would remove the "data" word. So `ImpersonatedUserNotExistsError` would be great IMO ;)

Reply:
Yeah, 404 is not the best choice - it's correct. But 403 can be questioned too. As I said, the user can be authorized to impersonate users but simply there is no such user. 

But it's a common problem of RESTfull services. HTTP codes are too narrow and operations are not resources (and pretending they are doesn't make things easier). 

That's why I said: let's do this if it makes our lives a little bit simpler.